File: Retos/Reto #21 - NÚMEROS PRIMOS GEMELOS [Media]/go/KevinED11.go
package main

import "fmt"

func isPrime(num int) bool {
	for n := 2; n <= num/2; n++ {
		if num < 2 || num%n == 0 {
			return false
		}
	}
	return true

}

func twinPrime(num int) (result [][]int) {
	for i := 2; i <= num; i++ {
		if isPrime(i) && isPrime(i+2) {
			result = append(result, []int{i, i + 2})
		}
	}

	return
}

func main() {
	num := 14
	twinPrimes := twinPrime(num)
	fmt.Println(twinPrimes)

}

File: Retos/Reto #21 - NÚMEROS PRIMOS GEMELOS [Media]/typescript/KevinED11.ts

type TwinPrime = [number, number]


const isPrime = (num: number): boolean => {
  if (num <= 1) return false
  for (let n = 2; n <= Math.sqrt(num); n++) {
    if (num % n === 0) return false
  } 

  return true
}


const twinPrime = (lnum: number): TwinPrime[] => {
  const numbers: TwinPrime[] = []
  let prevPrime = 2
  
  for (let n = 3; n <= Math.abs(lnum); n++) {
    if (isPrime(n) && (n - prevPrime) === 2){
      numbers.push([prevPrime, n])
    }
    prevPrime = isPrime(n) ? n : prevPrime
    
  }

  return numbers
    
  }
  

const mainF = () => {
  console.log(twinPrime(14))
  
}


mainF()